### PR TITLE
5086 Nav bar enhancement

### LINKED ISSUE #5086: Nav Bar Enhancement
## What type(s) of user does this feature affect?
 - volunteers
 - supervisors
 - admins
 

## Description
In an effort to improve the usability and overall feel of the app. We have made the decision to make a few changes to our navbar. Please refer to the [Figma file here](https://www.figma.com/file/iJ2vsIZHmNcAl5HPCfqvT4/Ruby-for-Good---CASA---Left-Navigation-Bar?type=design&node-id=0-1&mode=design&t=5LgOkxewS6NAW78f-0) and the acceptance criteria below - if you have any questions please ask @bcastillo32 (PM) and/or @MalloryPriceDesign (Design Lead). You can also message us on the casa slack channel. :) 

## Acceptance Criteria
**Hover Behavior and list item color**
Please reference [PlainAdmin's ](https://demo.plainadmin.com/icons) navbar - we want to use the same hover behavior and list item color as the PlainAdmin navbar

**Mobile (<770 px)**
**Closed State (default)**
1. The menu button is represented by a hamburger icon, with no text.
<br />

**Open State**
1. On click, the menu slides open from the left and a backdrop appears behind the menu.
2. At the top left of the menu is the hamburger button and on the top right is a horizontal logo.
- Logo updating behavior via the admin account should remain unchanged.
- The menu nav list items visual design and behavior is unchanged.
- Exiting the menu by clicking outside of the menu behavior should remain unchanged.
3. Clicking on the hamburger button in the open menu results in the menu sliding closed to the left.
<br />

**Desktop (>770 px)**
**Open State (default)**
1. The menu button is represented by a hamburger icon, with no text, and on the top right is a horizontal logo.
- Logo updating behavior via the admin account should remain unchanged.
- The menu nav list items visual design and behavior is unchanged.
2. Opening the nav bar displaces the site content to the right, this behavior is unchanged.
<br />

**Closed State**
1. Closing the nav bar results in a truncated nav bar, which includes the same hamburger button and the icons from the nav list.
2. Closing the nav bar expands the site content to the left, this behavior is unchanged.
<br />

**Hover State**
1. Hovering over the closed nav bar, starting at the list items and below, results in the nav bar expanding.
- This expansion does not displace site content.
2. Removing the mouse from hover, the nav bar returns to its closed state.
<br />

## How to access the QA site
_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by schoork**
### What github issue is this PR for, if any?
Resolves #5086 

### What changed, and why?
- Mobile:
  - button on bar makes collapsible/openable
- Desktop:
  - when minimized shows narrow bar with icons
  - when minimized and hovered over expands to full width

### How will this affect user permissions?
- Volunteer permissions: None
- Supervisor permissions: None
- Admin permissions: None

### How is this tested? (please write tests!) 💖💪
- components
  - spec/components/sidebar/group_component_spec.rb
  - spec/components/sidebar/link_component_spec.rb
- system
  - spec/system/case_contacts/index_spec.rb
  - spec/system/case_contacts/index_spec.rb
- views
  - spec/views/layouts/sidebar.html.erb_spec.rb

### Screenshots please :)
Screenshot 2023-10-09 at 7 34 30 PM
Screenshot 2023-10-09 at 7 34 25 PM
Screenshot 2023-10-09 at 7 32 00 PM
Screenshot 2023-10-09 at 7 31 55 PM
Screenshot 2023-10-09 at 7 31 50 PM

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:
`alt text`

### Feedback please? (optional)
We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

### CODE REVIEW

**Comment by schoork** [comment]
There are gonna be merge conflicts here between this and #5272. I can clean them up depending on the order they are merged.

File: app/assets/stylesheets/base/variables.scss [hunk 1]
```diff
@@ -1,2 +1,7 @@
 $primary: #00447c;
 $red: #dc3545;
+
+// ======== Sidebar colors
+$sidebar-inactive: #9AA4CA;
+$sidebar-active: #365CF5;
+$sidebar-dark: #1A2142;
```
Comment (javiguerra777) [inline app/assets/stylesheets/base/variables.scss:7]: I like the use of creating variables for these colors. Good job!

**Review by littleforest — COMMENTED** [review]
@schoork thanks so much for working on this! I'm getting really jumpy behavior on the hamburger menu on the small screen. Are you seeing differently?


https://github.com/rubyforgood/casa/assets/1938665/48ee1227-024d-43f3-9e07-d141804bf22c

**Comment by schoork** [comment]
@littleforest I did when I used the Chrome tools to get a smaller screen and did not refresh after. Try refreshing the page and it should go away. I felt like that was ignorable edge case, but could do some more work if you think it's worth supporting.

**Comment by schoork** [comment]
No, @littleforest, you're totally right. `window.screen.width` literally gets the size of your screen, whereas `winow.innerWidth` gets the size of the browser window. Good catch!

**Review by littleforest — APPROVED** [review]
@schoork this is such an improvement, thank you so much for working on this! The stakeholders will be so happy.

**Pull request merged**